File: src/main/java/com/matag/admin/stats/StatsController.java
package com.matag.admin.stats;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import lombok.AllArgsConstructor;

@RestController
@RequestMapping("/stats")
@AllArgsConstructor
public class StatsController {
  private final StatsService statsService;

  @GetMapping
  public StatsResponse stats() {
    return StatsResponse.builder()
      .totalUsers(statsService.countTotalUsers())
      .onlineUsers(statsService.onlineUsers())
      .totalCards(statsService.countCards())
      .totalSets(statsService.countSets())
      .build();
  }
}

File: src/main/kotlin/com/matag/admin/stats/StatsController.kt
package com.matag.admin.stats

import org.springframework.security.access.prepost.PreAuthorize
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/stats")
open class StatsController(
    private val statsService: StatsService
) {

    @PreAuthorize("permitAll()")
    @GetMapping
    open fun stats(): StatsResponse {
        return StatsResponse(
            totaNonGuestlUsers = statsService.countTotalUsers() - 1,
            onlineUsers = statsService.onlineUsers(),
            totalCards = statsService.countCards(),
            totalSets = statsService.countSets()
        )
    }
}
